File: src/priv/bajdcc/LALR1/grammar/runtime/RuntimeInstBase.java
package priv.bajdcc.LALR1.grammar.runtime;

/**
 * 【中间代码】指令结构基类
 *
 * @author bajdcc
 */
public abstract class RuntimeInstBase {

	public RuntimeInst inst = RuntimeInst.inop;

	public abstract int getAdvanceLength();

	public String toString(String delim) {
		return inst.toString();
	}
	
	@Override
	public String toString() {
		return toString(" ");
	}
	
	public abstract void gen(ICodegenByteWriter writer);
}

File: src/main/java/com/bajdcc/LALR1/grammar/runtime/RuntimeInstBase.kt
package com.bajdcc.LALR1.grammar.runtime

/**
 * 【中间代码】指令结构基类
 *
 * @author bajdcc
 */
abstract class RuntimeInstBase {

    var inst = RuntimeInst.inop

    abstract val advanceLength: Int

    open fun toString(delim: String): String {
        return inst.toString()
    }

    override fun toString(): String {
        return toString(" ")
    }

    abstract fun gen(writer: ICodegenByteWriter)
}
